cc/tf/rttops/math.cc

```diff
@@ -484,6 +484,54 @@ class RttMatMulOp : public OpKernel {
   bool transpose_b_;
 };
 
+template <typename Device>
+class RttMatMulAddOp : public OpKernel {
+ public:
+  explicit RttMatMulAddOp(OpKernelConstruction* context) : OpKernel(context) {
+  }
+
+  void Compute(OpKernelContext* context) override {
+    PRINT_BEG("RttMatMulAdd");
+
+    // Check if the dimensions of the two matrices are valid
+    const Tensor& x = context->input(0);
+    const Tensor& y = context->input(1);
+
+    cout << "input1: (" << x.dim_size(0) << "," << x.dim_size(1) << ")"
+         << ", dims: " << x.dims() << endl;
+    cout << "input2: (" << y.dim_size(0) << "," << y.dim_size(1) << ")"
+         << ", dims: " << y.dims() << endl;
+    OP_REQUIRES(
+      context, TensorShapeUtils::IsMatrix(x.shape()),
+      errors::InvalidArgument(
+        "In[0] is not a matrix. Instead it has shape ", x.shape().DebugString()));
+    OP_REQUIRES(
+      context, TensorShapeUtils::IsMatrix(y.shape()),
+      errors::InvalidArgument(
+        "In[1] is not a matrix. Instead it has shape ", y.shape().DebugString()));
+
+    OP_REQUIRES(
+      context, x.dim_size(1) == y.dim_size(0) && x.dim_size(0) == y.dim_size(0) && x.dim_size(1) == y.dim_size(1),

```

Comment:
In this functionality, You should also assert that the first input and the second input are both of the shape N*N.